modules/sdk-core/src/account-lib/mpc/tss/eddsa/eddsa.ts

```diff
@@ -283,35 +284,48 @@ export default class Eddsa {
       R: bigIntToBufferLE(R, 32).toString('hex'),
     };
 
+    const c = Eddsa.curve.basePointMult(split_r[jShare.i]);
+    const z = createHash('sha256').update(message).digest();
+
     const resultShares: SignShare = {
       xShare: P_i,
-      rShares: {},
-    };
-
-    for (let ind = 0; ind < jShares.length; ind++) {
-      const S_j = jShares[ind];
-      resultShares.rShares[S_j.i] = {
-        i: S_j.i,
+      commitment: {
+        c: bigIntToBufferLE(c, 32).toString('hex'),
+        z: z.toString('hex'),
+      },
+      rShare: {
+        i: jShare.i,
         j: pShare.i,
-        u: bigIntToBufferLE(split_u[S_j.i], 32).toString('hex'),
+        u: bigIntToBufferLE(split_u[jShare.i], 32).toString('hex'),
         v: bigIntToBufferLE(v[0], 32).toString('hex'),
-        r: bigIntToBufferLE(split_r[S_j.i], 32).toString('hex'),
+        r: bigIntToBufferLE(split_r[jShare.i], 32).toString('hex'),
         R: bigIntToBufferLE(R, 32).toString('hex'),
-      };
-    }
+      },
+    };
     return resultShares;
   }
 
-  sign(message: Buffer, playerShare: XShare, rShares: RShare[], yShares: YShare[] = []): GShare {
+  sign(
+    message: Buffer,
+    playerShare: XShare,
+    otherPlayerCommitment: Commitment,
```

Comment:
Adding a little colour on what this otherPlayerCommitment would be helpful in reviewers being able to understand what is going on here - but as someone not too in the weeds on this changes all look ok!